pkg/controller/injector/k8s_binding_injector_test.go

```diff
@@ -17,11 +17,175 @@ limitations under the License.
 import (
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	"github.com/kubernetes-incubator/service-catalog/pkg/brokerapi"
+	"k8s.io/client-go/1.5/kubernetes/fake"
+	kapi "k8s.io/kubernetes/pkg/api"
 	"testing"
 )
 
-func TestInject(t *testing.T) {
+func TestInjectOne(t *testing.T) {
+	binding, _ := getBindings()
+	cred, _ := getCreds()
+	injector := fakeK8sBindingInjector()
+	inject(t, injector, binding, cred)
+
+	secretsCl := injector.client.Core().Secrets(binding.Namespace)
+	secret, err := secretsCl.Get(binding.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred)
+}
+
+func TestInjectTwo(t *testing.T) {
+	binding0, binding1 := getBindings()
+	cred0, cred1 := getCreds()
+
+	injector := fakeK8sBindingInjector()
+	inject(t, injector, binding0, cred0)
+	inject(t, injector, binding1, cred1)
+
+	secretsCl := injector.client.Core().Secrets(binding0.Namespace)
+	secret, err := secretsCl.Get(binding0.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred0)
+
+	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
+	secret, err = secretsCl.Get(binding1.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred1)
+}
+
+func TestUninjectOne(t *testing.T) {
+	binding, _ := getBindings()
+	cred, _ := getCreds()
+
+	injector := fakeK8sBindingInjector()
+	inject(t, injector, binding, cred)
+	injector.Uninject(binding)
+
+	secretsCl := injector.client.Core().Secrets(binding.Namespace)
+	secret, err := secretsCl.Get(binding.Name)
+	if err == nil {
+		testCredentialsUninjected(t, secret.Data)
+	}
 }
 
-func TestUninject(t *testing.T) {
+func TestUninjectTwo(t *testing.T) {
+	binding0, binding1 := getBindings()
+	cred0, cred1 := getCreds()
+
+	injector := fakeK8sBindingInjector()
+	inject(t, injector, binding0, cred0)
+	inject(t, injector, binding1, cred1)
+
+	injector.Uninject(binding0)
+
+	// test that binding0 is gone
+	secretsCl := injector.client.Core().Secrets(binding0.Namespace)
+	secret, err := secretsCl.Get(binding0.Name)
+	if err == nil {
+		testCredentialsUninjected(t, secret.Data)
+	}
+
+	//test that binding1 is still there
+	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
+	secret, err = secretsCl.Get(binding1.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred1)
+
+	// test that binding1 is gone after uninject
+	injector.Uninject(binding1)
+
+	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
+	secret, err = secretsCl.Get(binding1.Name)
+	if err == nil {
+		testCredentialsUninjected(t, secret.Data)
+	}
+}
+
+func fakeK8sBindingInjector() *k8sBindingInjector {
+	return &k8sBindingInjector{
+		client: fake.NewSimpleClientset(),
+	}
+}
+
+func getBindings() (*servicecatalog.Binding, *servicecatalog.Binding) {
```

Comment:
This and the getCreds() funcs feel a little hard-coded to me w.r.t. their list of return values. I'm wondering what you think about instead of returning a fix list of 2 things we return `[]*servicecatalog.Binding` and the func takes the number of bindings the caller wants?